pennylane/hf/observable.py

```diff
@@ -0,0 +1,327 @@
+    >>> f = (coeffs, ops)
+    >>> print(qubit_operator(f))
+    ((-1+0j)) [Z0]
+    + ((1+0j)) [I0]
+    """
+    ops = []
+    coeffs = anp.array([])
+
+    for n, t in enumerate(o_ferm[1]):
+        if len(t) == 0:
+            ops = ops + [qml.Identity(0)]
+            coeffs = anp.array([0.0])
+            coeffs = coeffs + o_ferm[0][n]
+        else:
+            op = jordan_wigner(t)
+            if op != 0:
+                ops = ops + op[1]
+                coeffs = anp.concatenate([coeffs, anp.array(op[0]) * o_ferm[0][n]])
+
+    o_qubit = simplify(qml.Hamiltonian(coeffs, ops), cutoff=cutoff)
+
+    return o_qubit
+
+
+def jordan_wigner(op):
+    r"""Convert a fermionic operator to a qubit operator using the Jordan-Wigner mapping.
+
+    The one-body fermionic operator :math:`a_2^\dagger a_0` is constructed as [2, 0] and the
+    two-body operator :math:`a_4^\dagger a_3^\dagger a_2 a_1` is constructed as [4, 3, 2, 1].
+
+    Args:
+        op (list[int]): the fermionic operator
+
+    Returns
+        tuple(list[complex], list[list[int, str]]): list of coefficients and the qubit-operator terms
+
+    **Example**
+
+    >>> f  = [0, 0]
+    >>> q = jordan_wigner(f)
+    >>> q
+    ([(0.5+0j), (-0.5+0j)], [[], [(0, 'Z')]]) # corresponds to :math:`\frac{1}{2}(I_0 - Z_0)`
+    """
+    if len(op) == 1:
+        op = [((op[0], 1),)]
+
+    if len(op) == 2:
+        op = [((op[0], 1), (op[1], 0))]
+
+    if len(op) == 4:
+        op = [((op[0], 1), (op[1], 1), (op[2], 0), (op[3], 0))]
+
+        if op[0][0][0] == op[0][1][0] or op[0][2][0] == op[0][3][0]:
+            return 0
+
+    for t in op:
+        for l in t:
+            z = [(index, "Z") for index in range(l[0])]
+            x = z + [(l[0], "X"), 0.5]
+
+            if l[1]:
+                y = z + [(l[0], "Y"), -0.5j]
+
+            else:
+                y = z + [(l[0], "Y"), 0.5j]
+
+            if t.index(l) == 0:
+                q = [x, y]
+            else:
+                m = []
+                for t1 in q:
+                    for t2 in [x, y]:
+                        q1, c1 = _pauli_mult(t1[:-1], t2[:-1])
+                        m.append(q1 + [c1 * t1[-1] * t2[-1]])
+                q = m
+
+    c = [p[-1] for p in q]
+    o = [p[:-1] for p in q]
+
+    for item in o:
+        k = [i for i, x in enumerate(o) if x == item]
+        if len(k) >= 2:
+            for j in k[::-1][:-1]:
+                del o[j]
+                c[k[0]] = c[k[0]] + c[j]
+                del c[j]
+
+    pauli_map = {"X": qml.PauliX, "Y": qml.PauliY, "Z": qml.PauliZ}
+    for i, term in enumerate(o):
+        if len(term) == 0:
+            o[i] = qml.Identity(0)
+        if len(term) == 1:
+            o[i] = pauli_map[term[0][1]](term[0][0])
+        if len(term) > 1:
+            k = [pauli_map[t[1]](t[0]) for t in term]
+            o[i] = reduce(lambda x, y: x @ y, k)
+
+    return c, o
+
+
+def simplify(h, cutoff=1.0e-12):
+    r"""Add together identical terms in the Hamiltonian.
```

Comment:
@soranjh this one should go in `pennylane/ops/functions/simplify.py`, and imported top-level as `qml.simplify`